FirebaseAuth/Sources/Swift/Auth/Auth.swift

```diff
@@ -1723,21 +1723,21 @@ extension Auth: AuthInterop {
 
   #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
     private func addProtectedDataDidBecomeAvailableObserver() {
-      weak var weakSelf = self
       protectedDataDidBecomeAvailableObserver =
         NotificationCenter.default.addObserver(
           forName: UIApplication.protectedDataDidBecomeAvailableNotification,
           object: nil,
           queue: nil
-        ) { notification in
-          let strongSelf = weakSelf
-          if let observer = strongSelf?.protectedDataDidBecomeAvailableObserver {
+        ) { [weak self] notification in
+          guard let self else { return }
+          if let observer = self.protectedDataDidBecomeAvailableObserver {
             NotificationCenter.default.removeObserver(
               observer,
               name: UIApplication.protectedDataDidBecomeAvailableNotification,
               object: nil
             )
           }
+          self.protectedDataInitialization()
```

Comment:
Everything else in the PR should be a no-op. This this line restores 10.x behavior.